### PR TITLE
Remove support for relative path starting with /

### PR DESCRIPTION
**Pull request by pakrym**
https://github.com/aspnet/FileSystem/issues/104

@davidfowl @cesarbs

### CODE REVIEW

**Comment by pranavkm** [comment]
You might also want to try this with Mvc. We pass paths starting with / all the time to the file system.

**Comment by pranavkm** [comment]
Might also want to do something similar with `EmbeddedFileProvider`

**Comment by davidfowl** [comment]
This also requires an announcement

File: test/Microsoft.AspNet.FileProviders.Physical.Tests/PhysicalFileProviderTests.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public void ExistingFilesReturnTrue()
             Assert.NotNull(info);
             Assert.True(info.Exists);
 
-            info = provider.GetFileInfo("/File.txt");
```
Comment (davidfowl) [inline test/Microsoft.AspNet.FileProviders.Physical.Tests/PhysicalFileProviderTests.cs]: Need a negative test making sure that passing `/file` does the appropriate thing on each OS.

**Comment by pakrym** [comment]
Made same changes to embedded file system and added negative tests

**Comment by muratg** [comment]
@pakrym What's the behavior for Windows absolute paths such as `c:\foo\bar`?

**Comment by pakrym** [comment]
We returning NoopFileInfo

File: src/Microsoft.AspNet.FileProviders.Embedded/EmbeddedFileProvider.cs [hunk 2]
```diff
@@ -105,12 +96,6 @@ public IDirectoryContents GetDirectoryContents(string subpath)
...
             }
 
-            // Relative paths starting with a leading slash okay
-            if (subpath.StartsWith("/", StringComparison.Ordinal))
-            {
-                subpath = subpath.Substring(1);
-            }
-
             // Non-hierarchal.
             if (!subpath.Equals(string.Empty))
```
Comment (pranavkm) [inline src/Microsoft.AspNet.FileProviders.Embedded/EmbeddedFileProvider.cs]: Could you combine this with the condition at Line 94?

File: test/Microsoft.AspNet.FileProviders.Physical.Tests/PhysicalFileProviderTests.cs [hunk 3]
```diff
@@ -705,20 +722,16 @@ public void Tokens_With_Forward_And_Backward_Slash()
             using (var root = new DisposableFileSystem())
             {
                 var provider = new PhysicalFileProvider(root.RootPath);
-                var token1 = provider.Watch("/a/b");
-                var token2 = provider.Watch("a/b");
-                var token3 = provider.Watch(@"a\b");
+                var token1 = provider.Watch("a/b");
```
Comment (pranavkm) [inline test/Microsoft.AspNet.FileProviders.Physical.Tests/PhysicalFileProviderTests.cs]: Could you make this into a `Theory`?

**Comment by davidfowl** [comment]
We decided to punt this and deal with the inconsistency.

**Pull request closed without merging**